x/ibc/02-client/keeper/client.go

```diff
@@ -72,13 +86,18 @@ func (k Keeper) UpdateClient(ctx sdk.Context, clientID string, header exported.H
 	}
 
 	var (
-		consensusState exported.ConsensusState
-		err            error
+		consensusState      exported.ConsensusState
+		err                 error
+		oldHeaderNextValSet *tmtypes.ValidatorSet // TODO: should I get this from the msg or internal query?
 	)
 
 	switch clientType {
 	case exported.Tendermint:
-		clientState, consensusState, err = tendermint.CheckValidityAndUpdateState(clientState, header, ctx.ChainID())
+		unbondingTime := k.stakingKeeper.UnbondingTime(ctx)
```

Comment:
Why are you grabbing the unbonding time of the current chain? I think you want to grab the unbonding time of the connecting chain, which should be part of the client state. This is a change that will be added in the ICS-7 fixes (hope to start this tomorrow)

I think for now, we can just fill this in with some constant. And then when the ICS-7 fixes get merged, we can correct it

Reply:
> Why are you grabbing the unbonding time of the current chain.  I think you want to grab the unbonding time of the connecting chain

ahhh true

Reply:
This is correct, this should be the unbonding time of the chain we are verifying.